server/src/user-house/test/user-house.e2e-spec.ts

```diff
@@ -88,4 +88,231 @@ describe('UserHouseController (e2e)', () => {
+      // Log in as an admin
+      const loginUser = await request(app.getHttpServer())
+        .post('/auth/login')
+        .send({
+          email: 'alice@example.com',
+          password: 'password',
+        });
+
+      token = await loginUser.body.accessToken;
+
+      await request(app.getHttpServer())
+        .put('/user-houses/admin')
+        .send({
+          userId, // Assuming this user matches the loggedIn user
+          houseId,
+          isAdmin: false,
+        })
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(400);
+    });
+
+    it('should return 200 and change admin status to true', async () => {
+      // Log in as an admin
+      const loginUser = await request(app.getHttpServer())
+        .post('/auth/login')
+        .send({
+          email: 'diana@example.com',
+          password: 'password',
+        });
+
+      token = await loginUser.body.accessToken;
+
+      const houseId = 109; // Assuming this house is the logged-in user house
+      const userId = 105; // Assuming this user is the member of the logged-in user house
+
+      const response = await request(app.getHttpServer())
+        .put('/user-houses/admin')
+        .send({
+          userId,
+          houseId,
+          isAdmin: true,
+        })
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(200);
+
+      expect(response.body.isAdmin).toBe(true);
+    });
+  });
+
+  describe('DELETE /user-houses/:houseId', () => {
+    const houseId = 106; // Default houseId for the tests
+    const userId = 101; // Default userId for the tests
+
+    it('should return 401 if not authenticated', async () => {
+      await request(app.getHttpServer())
+        .delete(`/user-houses/${houseId}`)
+        .send({
+          userId,
+        })
+        .expect(401);
+    });
+
+    it('should return 403 if user is not the admin', async () => {
+      // Log in as a non-admin user
+      const loginUser = await request(app.getHttpServer())
+        .post('/auth/login')
+        .send({
+          email: 'bob@example.com',
+          password: 'password',
+        });
+
+      token = await loginUser.body.accessToken;
+
+      const houseId = 107; // Logged-in user house
+
+      await request(app.getHttpServer())
+        .delete(`/user-houses/${houseId}`)
+        .send({
+          userId,
+          houseId,
+          isAdmin: false,
+        })
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(403);
+    });
+
+    it('should return 404 if userHouse does not exist', async () => {
+      // Log in as an admin
+      const loginUser = await request(app.getHttpServer())
+        .post('/auth/login')
+        .send({
+          email: 'alice@example.com',
+          password: 'password',
+        });
```

Comment:
I tried to reduce the numbers of the logging in with different users, but still had to log in sometimes since the defaut user is overwritted soon after logging in with another user.
But I changed the default user to Diana. Same for admin status api! And I keep the same order as how the api works!
https://github.com/HAK2024/fairy-share/pull/90/commits/2c6afcb427bfb3f1ff8752843337c0184500722b